src/main/java/org/usfirst/frc/team2473/robot/subsystems/SparkDriveSubsystem.java

```diff
@@ -0,0 +1,250 @@
+				double newPower = power/tempTable.get(power);
+				return newPower;
+			}
+			
+			// linearize between the two values around the power
+			int i;
+			for (i = powers.size() - 1; powers.get(i) > power; i--);
+		
+			double lowerNearestPower = powers.get(i); //the largest power value that is lower than the power input in the lookup table
+			double higherNearestPower = powers.get(i+1); //the smallest power value that is greater than the power input in the lookup table
+			
+			double calibrationRatioOfLowerPower = tempTable.get(lowerNearestPower); 
+			double calibrationRatioOfHigherPower = tempTable.get(higherNearestPower); 	
+			
+			double slope = (calibrationRatioOfHigherPower-calibrationRatioOfLowerPower) / (higherNearestPower-lowerNearestPower);
+			
+			double deltaPower = power - lowerNearestPower; //the change of lookup ratio between the two bounds
+			
+			double powerCalibration = calibrationRatioOfLowerPower + slope * deltaPower;
+			
+			return power/powerCalibration;
+		}
+	}*/
+	
+	/**
+	 * Moves the robot with the given speed and rotation values in teleop mode.
+	 * @param speed		the robot's speed along the x-axis. Forward is positive.
+	 * @param rotation	the robot's rotation rate along the z-axis. Clockwise is positive.
+	 */
+	public void teleopDrive(double speed, double rotation) {
+		teleopDrive.arcadeDrive(speed, rotation);
+	}
+	
+	/**
+	 * Sets motor powers using the lookup table.
+	 * @param bl 	back left motor power
+	 * @param fl 	front left motor power
+	 * @param br 	back right motor power
+	 * @param fr 	front right motor power
+	 */
+	public void drive(double l, double r) {
+		driveRawPower(l, r);
+	}
+	
+	/**
+	 * Sets motor powers without using the lookup table.
+	 * @param bl 	back left motor power
+	 * @param fl 	front left motor power
+	 * @param br 	back right motor power
+	 * @param fr 	front right motor power
+	 */
+	public void driveRawPower(double l, double r) {
+		leftSpark.set(l);
+		rightSpark.set(r);
+	}
+	
+	/**
+	 * Sets all motor powers to 0.
+	 */
+	public void stopMotors() {
+		drive(0, 0);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public void initDefaultCommand() {}
+	
+	/**
+	 * Gets the gyro heading.
+	 * @return heading of the robot (degrees). Initial position is 0, clockwise is positive.
+	 */
+	public double getGyroAngle() {
+		return Devices.getInstance().getNavXGyro().getAngle();
+	}
+	
+	/**
+	 * Gets the encoder ticks of a given Talon based on id. Forward is positive.
+	 * @param id Talon ID
+	 * @return encoder ticks
+	 */
+	public synchronized double getEncoderTicks(int id) {
+		// flip sign of the ticks if the motor is on the right
+		if (id == RobotMap.SPARK_L) return -leftSpark.getEncoderPosition();
+		else return rightSpark.getEncoderPosition();
+	}
+	
+	/**
+	 * Sets all encoder values to 0.
+	 */
+	public synchronized void resetEncoders() {
+		// cannot do right now
+		encoderResetComplete = true;
```

Comment:
Shouldn't this return false then?

Reply:
If we return `false`, the robot will wait until the encoders are fully reset. Until REV Robotics releases the new API version that allows for resetting encoders, we need to return `true` to allow the robot to run.